File: app/src/main/java/com/example/audiobookmark/BookmarkRoomDatabase.java
package com.example.audiobookmark;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;


@Database(entities = {Bookmark.class}, version = 1)
public abstract class BookmarkRoomDatabase extends RoomDatabase {
    private static volatile BookmarkRoomDatabase INSTANCE;

    public abstract BookmarkDao dao();

    static BookmarkRoomDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (BookmarkRoomDatabase.class) {
                if (INSTANCE == null) {
                    // Create database here
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            BookmarkRoomDatabase.class, "audio_bookmarks_db")
                            .build();
                }
            }
        }
        return INSTANCE;
    }
}

File: app/src/main/java/de/augmentedmind/stopit/BookmarkRoomDatabase.kt
package de.augmentedmind.stopit

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = arrayOf(Bookmark::class), version = 1)
abstract class BookmarkRoomDatabase : RoomDatabase() {

    abstract fun dao(): BookmarkDao

    companion object {
        @Volatile
        private var INSTANCE: BookmarkRoomDatabase? = null

        fun getDatabase(context: Context): BookmarkRoomDatabase {
            val tempInstance = INSTANCE
            if (tempInstance != null) {
                return tempInstance
            }

            synchronized(this) {
                val instance = Room.databaseBuilder(
                        context.applicationContext,
                        BookmarkRoomDatabase::class.java,
                        "audio_bookmarks_db"
                ).build()
                INSTANCE = instance
                return instance
            }
        }
    }
}